### PR TITLE
chore(ecs-patterns): fix the integration tests of queue processing fargate service

### LINKED ISSUE #28383: (aws-ecs-patterns): Integration testing of aws-ecs-patterns contains an invalid integration test
### Describe the bug

The integration testing of the aws-ecs-patterns has an invalid docker being used,

1. When executed the container state always fails with the following message - 
image

2. Looking at the docker and python file it never creates a webserver so the test on this location(https://github.com/aws/aws-cdk/blob/7c62d6809cbdeee251c8ff31e22db7d0ff85aba2/packages/%40aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-public.ts#L18) ideally should always fail as it asks the container to either respond with a localhost response which is not possible as there is no web server building in the docker image or give exit 1 which is exit with failure.

3. It is possible there are other issues with the integration testing but I have found these while working on the bug - #20706

### Expected Behavior

All integration tests should be perfectly aligned and work as expected.

### Current Behavior

Integration tests go into infinite loop until cloud formation tasks fail.

### Reproduction Steps

Create a new integration step using the QueueProcessingFargateService, using the same Docker and the index.py with no webserver and the docker should be using the CMD command.

### Possible Solution

1. Separate Docker file for separate fargate integration tests.
2. Queue processing is different from application load balancer.

### Additional Information/Context

_No response_

### CDK CLI Version

NA

### Framework Version

2.115

### Node.js Version

NA

### OS

NA

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by AnuragMohapatra**
1. Fixed the docker image to use Python 3.11 to stop the python3.6 out of date warning/error
2. Fixed the lambda container's entry point handler, causing the integration tests to go into an infinite loop.
3. Fixed/changed the public integration test to a health check test as the queue processing fargate service doesn't have a public-facing feature.

Closes #28383.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.ts [hunk 1]
```diff
@@ -6,22 +6,26 @@ import * as integ from '@aws-cdk/integ-tests-alpha';
...
   assignPublicIp: true,
   healthCheck: {
-    command: ['CMD-SHELL', 'curl -f http://localhost/ || exit 1'],
-    interval: Duration.seconds(6),
+    command: ['CMD-SHELL', 'cat /tmp/health_status | grep -q "1" || exit 1'],
+    interval: Duration.seconds(10),
     retries: 10,
   },
+  environment: {
+    CONTAINER_HEALTH_CHECK: CONTAINER_HEALTH_CHECK,
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.ts:24]: What's this variable for?

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/sqs-reader/index.py [hunk 2]
```diff
@@ -6,17 +6,27 @@
...
-            QueueUrl=queue_url,
-            WaitTimeSeconds=10,
-        )
-        if response and 'Messages' in response:
-            for msg in response['Messages']:
-                print(msg['Body'])
-            entries = [{'Id': x['MessageId'], 'ReceiptHandle': x['ReceiptHandle']} for x in response['Messages']]
-            client.delete_message_batch(QueueUrl=queue_url, Entries=entries)
+    try:
+        client = boto3.client('sqs', region_name='ap-southeast-2')
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/sqs-reader/index.py:10]:
```suggestion
        client = boto3.client('sqs')
```
Let's leave the default region on the final file.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/asset.d3bdcfdd660bfe705ef686f270e773fc19d7c57af72814ab86e6e1c9ac2f7535/index.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/python
+import os
+import boto3
+
+QUEUE_NAME = os.environ.get('QUEUE_NAME')
+print('QUEUE_NAME ' + QUEUE_NAME)
+
+if __name__ == '__main__':
+    try:
+        client = boto3.client('sqs', region_name='ap-southeast-2')
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/asset.d3bdcfdd660bfe705ef686f270e773fc19d7c57af72814ab86e6e1c9ac2f7535/index.py:10]:
```suggestion
        client = boto3.client('sqs')
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
A couple of nits.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.ts (hunk 1 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.ts:24]: Removed, thanks.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/sqs-reader/index.py (hunk 2 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/sqs-reader/index.py]: Removed, thanks.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/asset.d3bdcfdd660bfe705ef686f270e773fc19d7c57af72814ab86e6e1c9ac2f7535/index.py (hunk 3 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/asset.d3bdcfdd660bfe705ef686f270e773fc19d7c57af72814ab86e6e1c9ac2f7535/index.py]: Removed, thanks.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/aws-ecs-patterns-queue-health-check.template.json [hunk 4]
```diff
@@ -0,0 +1,948 @@
...
+  "HealthCheckQueueServiceQueueProcessingTaskDefE2DEF18C": {
+   "Type": "AWS::ECS::TaskDefinition",
+   "Properties": {
+    "ContainerDefinitions": [
+     {
+      "Environment": [
+       {
+        "Name": "CONTAINER_HEALTH_CHECK",
+        "Value": "0"
+       },
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/aws-ecs-patterns-queue-health-check.template.json:476]:
```suggestion
```
To fix failing build.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/tree.json [hunk 5]
```diff
@@ -0,0 +1,1621 @@
...
+                                "awslogs-region": {
+                                  "Ref": "AWS::Region"
+                                }
+                              }
+                            },
+                            "environment": [
+                              {
+                                "name": "CONTAINER_HEALTH_CHECK",
+                                "value": "0"
+                              },
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/tree.json:872]:
```suggestion
```
Should be removed as well.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
Just update the integration test snapshots and this will be good for me.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/tree.json (hunk 5 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/tree.json]: Fixed thanks.

File: packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/aws-ecs-patterns-queue-health-check.template.json (hunk 4 above)
Comment (AnuragMohapatra) [reply packages/@aws-cdk-testing/framework-integ/test/aws-ecs-patterns/test/fargate/integ.queue-processing-fargate-service-health-check.js.snapshot/aws-ecs-patterns-queue-health-check.template.json]: Fixed thanks.

**Review by kaizencc — APPROVED** [review]
(updated to chore because although this does fix something, it doesn't fix the public api :))

Thanks @AnuragMohapatra for taking this on and @lpizzinidev for reviewing. It looks like the changes you're adding are all reasonable + helpful, so I'm inclined to trust you that this is working now, or at least not worse than before :). Appreciate you taking on this integ test fix!

**Pull request merged**